source/auth/tests/mongodb-oidc.rst

```diff
@@ -130,6 +130,47 @@ credentials.
 #. Create a new client using the AWS device workflow.
 #. Ensure that a ``find`` operation does not add credentials to the cache.
 
+Speculative Authentication
+==========================
+
+#. Clear the cache.
+#. Create a client with a request callback that returns a valid token
+  that will not expire soon.
+#. Set a fail point for ``saslStart`` commands of the form:
+
+.. code:: javascript
+
+    {
+        "mode": {"times": 2},
+        "data": {"failCommands": ["saslStart"], "errorCode": 18},
+    }
+
+#. Perform a ``find`` operation.
+#. Close the client.
+#. Create a new client with the same properties.
+#. Set a fail point for ``saslStart`` commands.
+#. Perform a ``find`` operation.
+#. Close the client.
+
+#. Clear the cache.
+#. Create a client with a request callback that returns a valid token
+  that will expire soon.
+#. Set a fail point for ``saslStart`` commands of the form:
+
+.. code:: javascript
+
+    {
+        "mode": {"times": 2},
```

Comment:
let's add appName to ensure that unused configuration won't affect rest of tests

Reply:
also, let's sue the full form like in reauth example

Reply:
We have a different way of handling it it Python, updated.